core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -94,6 +94,13 @@ private[spark] class AppStatusStore(
       }.toSeq
   }
 
+  def getJobIdsAssociatedWithStage(stageId: Int): Seq[Set[Int]] = {
+    store.view(classOf[StageDataWrapper]).index("stageId").first(stageId).last(stageId)
```

Comment:
can we avoid the store look up here?

Reply:
I really did not get the question, perhaps you are suggesting to try an alternative way but is there any other alternative way to do this? Let me know your thoughts. Thank you.